Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -42,12 +66,64 @@ struct Info
   template <typename Enum,
             std::enable_if_t<std::is_same<T, detail::UnderlyingType<Enum>>::value>* = nullptr>
   Info(const Info<Enum>& other)
-      : location{other.location}, default_value{static_cast<detail::UnderlyingType<Enum>>(
-                                      other.default_value)}
+      : m_location{other.GetLocation()}, m_default_value{static_cast<T>(other.GetDefaultValue())},
+        m_cached_value{other.template GetCachedValueCasted<T>()}
+  {
+  }
+
+  Info<T>& operator=(const Info<T>& other)
+  {
+    m_location = other.GetLocation();
+    m_default_value = other.GetDefaultValue();
+    m_cached_value = other.GetCachedValue();
+  }
```

Comment:
`return *this;` -- Clang will happily see it's UB and assume codepath leading to this function is unreachable :)

Reply:
Why doesn't C++ give you an error when you forget to return :(

Reply:
Google suggests `-Wreturn-type` does that.